devito/passes/clusters/aliases.py

```diff
@@ -163,12 +166,44 @@ def _aliases_from_clusters(self, clusters, exclude, meta):
 
         return processed
 
+    def process(self, clusters):
+        raise NotImplementedError
+
+    def _generate(self, exprs, exclude):
+        """
+        Generate one or more extractions from ``exprs``. An extraction is a
+        set of CIRE candidates which may be turned into aliases. Two different
+        extractions may contain overlapping sub-expressions and, therefore,
+        should be processed and evaluated indipendently. An extraction won't
+        contain any of the symbols appearing in ``exclude``.
+        """
+        raise NotImplementedError
+
+    def _lookup_key(self, c):
+        """
+        Create a key for the given Cluster. Clusters with same key may be
+        processed together in the search for CIRE candidates. Clusters should
+        have a different key if they must not be processed together, e.g.,
+        when this would lead to violation of data dependencies.
+        """
+        raise NotImplementedError
+
+    def _select(self, variants):
+        """
+        Select the best variant out of a set of variants, weighing flops
+        and working set.
+        """
+        raise NotImplementedError
+
+
+class CireTransformerLegacy(CireTransformer):
+
     def _do_generate(self, exprs, exclude, cbk_search, cbk_compose=None):
         """
         Carry out the bulk of the work of ``_generate``.
         """
         counter = generator()
-        make = lambda: Symbol(name='dummy%d' % counter())
+        make = lambda: Symbol(name='dummy%d' % counter(), dtype=np.float32)
```

Comment:
it doesn't matter, as long as it's not int32 we're good. These symbols are just temporary change of variables.
We just don't want the default np.int32 so that `estimate_cost` returns a proper operation count